.github/workflows/pkgci_test_riscv64.yml

```diff
@@ -0,0 +1,109 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Workflow for RISCV64 cross-compilation and test jobs.
+# The concurrency of this workflow is controlled by the caller's job.
+
+name: PkgCI Test RISCV64
+
+on:
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      write-caches:
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      write-caches:
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  cross_compile:
+    runs-on: ubuntu-20.04
+    env:
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      VENV_DIR: ${{ github.workspace }}/.venv
+      PLATFORM: linux
+      ARCH: riscv_64
+      ABI: lp64d
+      TARGET_BUILD_DIR: build-linux-riscv_64
+      DOCKER_IMAGE: "gcr.io/iree-oss/emscripten@sha256:2dd4c52f1bb499ab365aad0111fe5538b685d88af38636b409b0cf6a576ab214"
+      IREE_READ_REMOTE_CCACHE: 0
+      IREE_WRITE_REMOTE_CCACHE: 0
+      IREE_READ_LOCAL_CCACHE: 1
+      IREE_WRITE_LOCAL_CCACHE: ${{ inputs.write-caches }}
+      BUILD_PRESET: test
+    steps:
+      # General setup.
+      - name: "Checking out repository"
+        uses: actions/checkout@v4.1.7
+      - name: "Checking out runtime submodules"
+        run: ./build_tools/scripts/git/update_runtime_submodules.sh
+      - uses: actions/setup-python@v5.1.0
+        with:
+          # Must match the subset of versions built in pkgci_build_packages.
+          python-version: "3.11"
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: ${{ github.job }}
+          save: ${{ inputs.write-caches == 1 }}
+      - uses: actions/download-artifact@v4.1.7
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+      - name: Setup base venv
+        run: |
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+      - name: Install build dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y ninja-build
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: r25b
+      - name: Build riscv64
+        env:
+          HOST_BIN_DIR: ${{ env.VENV_DIR }}/bin
+          BUILD_SCRIPT: "./build_tools/cmake/build_riscv.sh"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
```

Comment:
https://github.com/iree-org/iree/actions/runs/10448717693/job/28947859057?pr=18274#step:10:185
```
++ realpath /home/runner/work/iree/iree/.venv/bin
realpath: /home/runner/work/iree/iree/.venv/bin: No such file or directory
```

This is one of the issues with using Docker together with pkgci that I started debugging. The python virtual environment setup above is done outside of the container and the container isn't inheriting it. My workaround for other workflows was to stop using Docker entirely.

Other ideas:
* Inherit the virtual environment somehow
* Move the virtual environment setup into the build script
* Run the entire workflow in a container (https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container)